Sources/FluentUI_macOS/Core/ControlHostingView.swift

```diff
@@ -4,34 +4,71 @@
 //
 
 import AppKit
+import Combine
 import SwiftUI
 
-/// Common wrapper for hosting and exposing SwiftUI components to AppKit-based clients.
-open class ControlHostingView: NSHostingView<AnyView> {
+/// Common wrapper for hosting and exposing SwiftUI components to UIKit-based clients.
+open class ControlHostingView: NSView {
+
+	/// The intrinsic content size of the wrapped SwiftUI view.
+	@objc public override var intrinsicContentSize: CGSize {
+		// Our desired size should always be the same as our hosted view.
+		return hostingView.intrinsicContentSize
+	}
+
 	/// Initializes and returns an instance of `ControlHostingContainer` that wraps `controlView`.
 	///
 	/// Unfortunately this class can't use Swift generics, which are incompatible with Objective-C interop. Instead we have to wrap
 	/// the control view in an `AnyView.`
 	///
 	/// - Parameter controlView: An `AnyView`-wrapped component to host.
-	/// - Parameter safeAreaRegions: Passthrough to the respective property on NSHostingView.
+	/// - Parameter safeAreaRegions: Passthrough to the respective property on UIHostingController.
```

Comment:
Assuming this should still say NSHostingView

Reply:
Whoops, yeah copied a bit from the iOS one